### PR TITLE
sepolia deployment, tweaks in deployment scripts

### PR DESCRIPTION
**Pull request by jellegerbrandy**

### CODE REVIEW

File: deploy/009_verify_deployment.js [hunk 1]
```diff
@@ -29,6 +29,21 @@ const func = async function (hre) {
     const CANCELLER_ROLE = await read('HATTimelockController', {}, 'CANCELLER_ROLE');
     const EXECUTOR_ROLE = await read('HATTimelockController', {}, 'EXECUTOR_ROLE');
 
+    // prin some general info before diagnosing
```
Comment (ben-kaufman) [inline deploy/009_verify_deployment.js:32]:
```suggestion
    // print some general info before diagnosing
```

File: deploy/009_verify_deployment.js [hunk 2]
```diff
@@ -106,6 +114,42 @@ const func = async function (hre) {
...
+    );
+
+    // if unexpected roles were granted we print some extra info
+
+        if (logs.length > 4 + executors.length) {
+          console.log(`Expected ${4 + executors.length} roles, but ${logs.length} were found in the logs`);
+          const timelockAddress = (await deployments.get('HATTimelockController')).address;
+          
+          const EXPECTED_ROLES = {
+            [governance]: [PROPOSER_ROLE, CANCELLER_ROLE, EXECUTOR_ROLE],
```
Comment (ben-kaufman) [inline deploy/009_verify_deployment.js:135]: Governance isn't necessarily expected to have the executor role.

File: hardhat.config.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ require("hardhat-watcher");
 require("hardhat-gas-reporter");
 require("@openzeppelin/hardhat-upgrades");
 require("dotenv").config();
-require("@primitivefi/hardhat-dodoc");
+// require("@primitivefi/hardhat-dodoc");
```
Comment (ben-kaufman) [inline hardhat.config.js:11]: Why is this commented out?

File: deploy/001_deploy_hattimelockcontroller.js [hunk 4]
```diff
@@ -14,9 +14,9 @@ const func = async function (hre) {
     governance = deployer;
   }
 
-  let hatGovernanceDelay = config["timelockDelay"];
```
Comment (ben-kaufman) [inline deploy/001_deploy_hattimelockcontroller.js:17]: Unrelated to this line but the execute in line 38 shouldn't be done anymore with the new version of time lock controller 0f OZ 4.8.1

File: deploy/009_verify_deployment.js [hunk 5]
```diff
@@ -106,6 +114,42 @@ const func = async function (hre) {
         "TIMELOCK_ADMIN_ROLE should be the admin role of the EXECUTOR_ROLE"
     );
 
+    verify(
+      !(await read('HATTimelockController', {}, 'hasRole', TIMELOCK_ADMIN_ROLE, deployer)),
+      `TIMELOCK_ADMIN_ROLE should NOT be the admin role of the deployer ${deployer}`
+    );
+    // Roles granted should be the 4 + number of executors
+    // (renounced deployer role, timelock admin of itself, governance proposer and canceller roles, and executor role to the executors)
```
Comment (ben-kaufman) [inline deploy/009_verify_deployment.js:122]: Should now be 3 not 4 because deployer role is not set by default anymore with OZ 4.8.1

Comment (jellegerbrandy) [reply deploy/009_verify_deployment.js:122]: well it is not let me find out why :)

**Review by ben-kaufman — APPROVED** [review]

**Pull request merged**